packages/cli/cli/src/commands/generate-openapi-fdr/generateOpenApiToFdrApiDefinitionForWorkspaces.ts

```diff
@@ -0,0 +1,71 @@
+import { AbsoluteFilePath, stringifyLargeObject } from "@fern-api/fs-utils";
+import { Project } from "@fern-api/project-loader";
+import { writeFile } from "fs/promises";
+import path from "path";
+import { CliContext } from "../../cli-context/CliContext";
+import { getAllOpenAPISpecs, LazyFernWorkspace, OSSWorkspace } from "@fern-api/lazy-fern-workspace";
+// TODO: clean up imports
+import { OpenApiDocumentConverterNode } from "@fern-api/docs-parsers";
+import { ErrorCollector } from "@fern-api/docs-parsers";
+import fs from "fs";
+import yaml from "js-yaml";
+import { BaseOpenApiV3_1ConverterNodeContext } from "@fern-api/docs-parsers";
+import { OpenAPIV3_1 } from "openapi-types";
+
+export async function generateOpenApiToFdrApiDefinitionForWorkspaces({
+    project,
+    outputFilepath,
+    cliContext
+}: {
+    project: Project;
+    outputFilepath: AbsoluteFilePath;
+    cliContext: CliContext;
+}): Promise<void> {
+    await Promise.all(
+        project.apiWorkspaces.map(async (workspace) => {
+            await cliContext.runTaskForWorkspace(workspace, async (context) => {
+                await cliContext.runTaskForWorkspace(workspace, async (context) => {
+                    if (workspace instanceof LazyFernWorkspace) {
+                        context.logger.info("Skipping, API is specified as a Fern Definition.");
+                        return;
+                    } else if (!(workspace instanceof OSSWorkspace)) {
+                        return;
+                    }
+                    const openAPISpecs = await getAllOpenAPISpecs({ context, specs: workspace.specs });
+
+                    const openApi = openAPISpecs[0];
```

Comment:
can your parser handle taking more than one ? this is going to be important in most places

Reply:
if not, we should log an error if > 0 and track that work

Reply:
The underlying referencing is what would need to support multiple inputs -- this boils down to a few functions in the parser OR exposing a document merger. I think we should discuss this.

The intention is to support multiple files -- I need to understand how the files can be merged together a bit before implementation. I'll leave a TODO, an error, and track the work!